db-client/src/db.ts

```diff
@@ -69,7 +69,7 @@ export class DB {
       // it by referring to value in an object.
       const newValue = updater(deepFreeze({ value: value as T }).value);
       checkValue(newValue);
-      if (await this.setIfVersion(key, newValue, version)) return deepFreeze(newValue);
+      if (await this.setIfVersion(key, version, newValue)) return deepFreeze(newValue);
```

Comment:
[Sonarts](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/arguments-order.md) rule might prevent this. If type checking is not enough perhaps worth making them named instead of positional arguments.

Reply:
I changed the signature to look like the one in `interfaces/src/db.ts`.